### PR TITLE
added flag for clean columns for joining table dataset with source dataset

### PR DESCRIPTION
**Pull request by ywkim312**
@longshuicy @jonglee1 @navarroc There was already an method for joining the table dataset with source dataset. I just added another variable to clean up the columns. If the new variable is true, it will remove all the columns from source dataset except 'geometry' and 'guid'. So eventually. the output geodataframe will only have the columns from table dataset with geometry. The user can easily convert this GeoDataFrame to shapefile or geopakage using geopands. 

I am not very sure how we will handle this join process at this moment. We probably need to discuss more how to do it. However, having this PR will be helpful in some cases

### CODE REVIEW

File: pyincore/utils/datasetutil.py [hunk 1]
```diff
@@ -17,30 +17,38 @@
 
 class DatasetUtil:
     @staticmethod
-    def join_datasets(geodataset, tabledataset):
+    def join_datasets(geodataset, tabledataset, clean_attributes=True):
```
Comment (longshuicy) [inline pyincore/utils/datasetutil.py:20]: could we default to "false" to be backwards compatible?

**Review by longshuicy — CHANGES_REQUESTED** [review]

File: pyincore/utils/datasetutil.py (hunk 1 above)
Comment (ywkim312) [reply pyincore/utils/datasetutil.py]: changed to false

**Comment by ywkim312** [comment]
> @ywkim312 Good to know that we already have a method for that. Im most curious about how fast it is? Could you have some benchmark for the SLC data I shared?

@longshuicy Sure, would you be able to provide the dataset id for SLC data and the table data that I can test?

File: pyincore/utils/datasetutil.py [hunk 2]
```diff
@@ -17,30 +17,38 @@
...
+            gdf = gdf[['geometry','guid']]
+
+        # joining and indexing
         join_gdf = gdf.set_index("guid").join(df.set_index("guid"))
 
         return join_gdf
 
     @staticmethod
-    def join_table_dataset_with_source_dataset(dataset, client):
+    def join_table_dataset_with_source_dataset(dataset, client, clean_attributes=True):
```
Comment (longshuicy) [inline pyincore/utils/datasetutil.py:45]: So sorry I missed a spot, same here ...

Comment (ywkim312) [reply pyincore/utils/datasetutil.py:45]: no worries. updated

**Review by longshuicy — APPROVED** [review]

**Pull request merged**